File: Java/CardViewDashboard/app/src/main/java/com/dodemy/cardviewdashboard/MainActivity.java
package com.dodemy.cardviewdashboard;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.os.Bundle;
import android.view.MenuItem;

import com.google.android.material.bottomnavigation.BottomNavigationView;

import com.dodemy.cardviewdashboard.fragments.CustomFragment;
import com.dodemy.cardviewdashboard.fragments.HomeFragment;
import com.dodemy.cardviewdashboard.fragments.SpeakFragment;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //creating fragment transition
        FragmentManager fragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        //adding fragment to current activity
        Fragment fragment = new HomeFragment();
        fragmentTransaction.add(R.id.fragment_layout, fragment);
        fragmentTransaction.commit();

        BottomNavigationView bottomNav = findViewById(R.id.bottom_navigation);
        bottomNav.setOnNavigationItemSelectedListener(navlistener);
    }

    private BottomNavigationView.OnNavigationItemSelectedListener navlistener = new BottomNavigationView.OnNavigationItemSelectedListener() {
        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
            Fragment fragment = null;
            switch(menuItem.getItemId()) {
                case R.id.navigation_home:
                    fragment = new HomeFragment();
                    break;
                case R.id.navigation_custom:
                    fragment = new CustomFragment();
                    break;
                case R.id.navigation_dashboard:
                    fragment = new SpeakFragment();
                    break;
            }
            getSupportFragmentManager().beginTransaction().replace(R.id.fragment_layout,
                    fragment).commit();
            return true;
        }
    };
}

File: AndroidTesting/Kotlin/EspressoUITest/TestingFragmentsInIsolation/app/src/main/java/com/dodemy/testingfragmentsinisolation/ui/movie/MainActivity.kt
package com.dodemy.testingfragmentsinisolation.ui.movie

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.dodemy.testingfragmentsinisolation.factory.MovieFragmentFactory
import com.dodemy.testingfragmentsinisolation.R

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        supportFragmentManager.fragmentFactory = MovieFragmentFactory()
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        init()
    }

    private fun init(){
        if(supportFragmentManager.fragments.size == 0){
            val movieId = 1
            val bundle = Bundle()
            bundle.putInt("movie_id", movieId)
            supportFragmentManager.beginTransaction()
                .replace(R.id.container, MovieDetailFragment::class.java, bundle)
                .commit()
        }
    }

}







